contracts/CommonHealthCheck.sol

```diff
@@ -94,37 +73,42 @@ contract CommonHealthCheck {
     }
 
     function check(
+        address strategy,
         uint256 profit,
         uint256 loss,
         uint256 debtPayment,
         uint256 debtOutstanding,
         uint256 totalDebt
     ) external view returns (bool) {
-        return _runChecks(profit, loss, debtPayment, debtOutstanding, totalDebt);
+        require(strategy != address(0));
+
+        return _runChecks(strategy, profit, loss, debtPayment, debtOutstanding, totalDebt);
     }
 
     function _runChecks(
+        address strategy,
         uint256 profit,
         uint256 loss,
         uint256 debtPayment,
         uint256 debtOutstanding,
         uint256 totalDebt
     ) internal view returns (bool) {
-        address customCheck = checks[msg.sender];
+        address customCheck = checks[strategy];
 
         if (customCheck == address(0)) {
-            return _executeDefaultCheck(profit, loss, totalDebt);
+            return _executeDefaultCheck(strategy, profit, loss, totalDebt);
         }
 
-        return CustomHealthCheck(customCheck).check(profit, loss, debtPayment, debtOutstanding, msg.sender);
+        return ICustomHealthCheck(customCheck).check(strategy, profit, loss, debtPayment, debtOutstanding);
     }
 
     function _executeDefaultCheck(
+        address strategy,
         uint256 _profit,
         uint256 _loss,
         uint256 _totalDebt
     ) internal view returns (bool) {
-        Limits memory limits = strategiesLimits[msg.sender];
+        Limits memory limits = strategiesLimits[strategy];
         uint256 _profitLimitRatio;
         uint256 _lossLimitRatio;
         if (limits.exists) {
```

Comment:
we can add a check here based on  a mapping to skip the Next Health check one time, and is setup per strategy address and can be set by authorized. this will be reset back to false after this one time.

That way the disabling of a one time healthcheck logic lies in this contract and saves some space on vault.